### PR TITLE
Added testing for hipStreamSynchronize and hipStreamQuery

### PR DESCRIPTION
**Pull request by fabiomestre**
Note:

This PR currently relies on stream concurrency to perform the testing. This is not supported by HIP and an alternative solution is being worked on which is currently pending on the output of EXSWCPHIPT-78.
This PR will be updated once EXSWCPHIPT-78 is closed.

### CODE REVIEW

**Comment by fabiomestre** [comment]
This PR currently relies on stream concurrency to perform the testing. This is not supported by HIP and an alternative solution is being worked on which is currently pending on the output of EXSWCPHIPT-78.
This PR will be updated once EXSWCPHIPT-78 is closed.

**Review by chrispaquot — DISMISSED** [review]

**Comment by alistair-low** [comment]
Fabio will resolve the merge conflicts when he returns from holiday 😊

**Review by chrispaquot — DISMISSED** [review]

File: tests/catch/unit/stream/hipStreamSynchronize.cc [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+  if (stream == explicitStream) {
+    HIP_CHECK(hipStreamCreate(&stream));
+  }
+
+  HipTest::runKernelForDuration(std::chrono::milliseconds(500), stream);
+  HIP_CHECK(hipStreamSynchronize(stream));
+  HIP_CHECK(hipStreamQuery(stream));
+
+  if (stream != hip::nullStream && stream != hip::streamPerThread) {
```
Comment (FMarno) [inline tests/catch/unit/stream/hipStreamSynchronize.cc]:
```suggestion
  const bool isExplicitStream = stream == explicitStream;
  if (isExplicitStream) {
    HIP_CHECK(hipStreamCreate(&stream));
  }

  HipTest::runKernelForDuration(std::chrono::milliseconds(500), stream);
  HIP_CHECK(hipStreamSynchronize(stream));
  HIP_CHECK(hipStreamQuery(stream));

  if (isExplicitStream) {
```

To make it more obvious that these are testing the same condition

**Review by FMarno — DISMISSED** [review]

**Comment by fabiomestre** [comment]
@chrispaquot This PR has been updated. It can be merged once https://github.com/ROCm-Developer-Tools/HIP/pull/2750 is merged

**Pull request merged**